### DISCUSSION TITLE
Tweaking flocking model to torus space?

### DISCUSSION THREAD

**Discussion by selforganizing** [JoelForamitti/agentpy#60] [Q&A]
Great package, really useful. I'm trying to tweak the (Reynolds) flocking model to have a toroidal space, instead of avoiding the spatial borders.  But the boids get stuck on the borders (see screenshot below) instead of flying through to the other side? I commented out the code segment calculating v4 (and omitted it from the velocity addition) and switched `torus=True` in the space declaration. What do you think am I missing? Thanks a bunch in advance ~
![Screenshot (82)](https://user-images.githubusercontent.com/107113107/203285473-cb07246a-7d54-4058-991b-80628d174e2f.jpg)

**Comment by jofmi** [comment]
Hi @selforganizing ! I am not sure myself, I had followed the logic from this pseudocode: https://vergenet.net/~conrad/boids/pseudocode.html

If you find a better logic for this model, feel invited to suggest a change to the model :)
